Nodejs/Product/TestAdapterNetStandard/NetCoreTestDiscoverer.cs

```diff
@@ -70,73 +72,43 @@ public void DiscoverTests(IEnumerable<string> sources, IDiscoveryContext discove
 
                 foreach (var (projectFile, propertyGroup) in projects)
                 {
-                    var testFramework = propertyGroup.Descendants(NodeProjectProperty.TestFramework).FirstOrDefault()?.Value;
-                    var testRoot = propertyGroup.Descendants(NodeProjectProperty.TestRoot).FirstOrDefault()?.Value;
-                    var outDir = propertyGroup.Descendants(NodeProjectProperty.TypeScriptOutDir).FirstOrDefault()?.Value ?? "";
-
-                    if (string.IsNullOrEmpty(testRoot) || string.IsNullOrEmpty(testFramework))
-                    {
-                        logger.SendMessage(TestMessageLevel.Warning, $"No TestRoot or TestFramework specified for '{Path.GetFileName(projectFile)}'.");
-                        continue;
-                    }
-
                     var projectHome = Path.GetDirectoryName(projectFile);
-                    var testItems = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
-                    var testFolder = Path.Combine(projectHome, testRoot);
-
-                    if (!Directory.Exists(testFolder))
-                    {
-                        logger.SendMessage(TestMessageLevel.Warning, $"Test folder path '{testFolder}' doesn't exist.");
-                        continue;
-                    }
-
-                    // grab all files, we try for .ts files first, and only parse the .js files if we don't find any
-                    foreach (var file in Directory.EnumerateFiles(testFolder, "*.ts", SearchOption.AllDirectories))
-                    {
-                        ProcessFiles(file);
-                    }
 
+                    // Prioritize configuration files over manually setup tests files.
+                    var testItems = this.GetConfigItems(projectHome);
                     if (!testItems.Any())
                     {
-                        foreach (var file in Directory.EnumerateFiles(testFolder, "*.js", SearchOption.AllDirectories))
+                        var testFramework = propertyGroup.Descendants(NodeProjectProperty.TestFramework).FirstOrDefault()?.Value;
+                        var testRoot = propertyGroup.Descendants(NodeProjectProperty.TestRoot).FirstOrDefault()?.Value;
+                        var outDir = propertyGroup.Descendants(NodeProjectProperty.TypeScriptOutDir).FirstOrDefault()?.Value ?? "";
```

Comment:
Personally, I'd put the `?? ""` at the use site, to maintain the parallelism.

Reply:
Actually is already handled downstream. So I have removed it.